### PR TITLE
Fixed internal error raised for incomplete udev rule in AY profile

### PR DESCRIPTION
**Pull request by mchf**
bnc#1020074

### CODE REVIEW

File: src/lib/network/network_autoyast.rb [hunk 1]
```diff
@@ -228,7 +237,11 @@ def assign_udevs_to_devs(udev_rules)
       udev_rules.each do |rule|
         name_to = rule["name"]
         attr = rule["rule"]
-        key = rule["value"].downcase
+        key = rule["value"]
+
+        next if !valid_rename_udev_rule?(rule)
```
Comment (mvidner) [inline src/lib/network/network_autoyast.rb]: You can keep the original line if you move this one before it.

File: test/network_autoyast_test.rb [hunk 2]
```diff
@@ -319,4 +319,33 @@ def keep_install_network_value(value)
       end
     end
   end
+
+  describe "#valid_rename_udev_rule?" do
+    it "fails when the rule do not contain new name" do
+      rule = { "rule" => "exists", "value" => "as_well" }
```
Comment (mvidner) [inline test/network_autoyast_test.rb]: Please use meaningful values when it does not hinder the test. Copying the values from the bug should do it.
>        <rule>
>          <name>eth0</name>
>          <rule>ATTR{address}</rule>
>          <value>00:02:29:69:d3:63</value>
>        </rule>

**Comment by mvidner** [comment]
From a high level perspective, I dislike that this logic is added to the AutoYaST part of the code, but it checks the internal consistency of a udev rule. Instead, we should have a udev rule model class that has a `from_ay` constructor and a `valid?` method. But that may not be appropriate for an SP2 update.

**Comment by mchf** [comment]
> From a high level perspective, I dislike that this logic is added to the AutoYaST part of the code, but it checks the internal consistency of a udev rule.

depends. I've though of it as a kind of "user input check". Mostly bcs it checks requirements we have for udev read from a profile. Generic udev rule check should be more detailed.

File: test/network_autoyast_test.rb (hunk 2 above)
Comment (mchf) [inline test/network_autoyast_test.rb]: well the bug was about missing attributes ;-) but ok, I'll use these meaningful values.

File: src/lib/network/network_autoyast.rb (hunk 1 above)
Comment (mchf) [inline src/lib/network/network_autoyast.rb]: i know. This was a kind of cosmetic decision. Is it blocker?

**Comment by mchf** [comment]
Thanks for review

**Pull request merged**